File: app/src/main/java/com/pdamkotasmg/happywork/api/server/ApiService.java
package com.pdamkotasmg.happywork.api.server;

import com.pdamkotasmg.happywork.fitur.presensi.model.checkLocationModel.CheckLocationRootModel;
import com.pdamkotasmg.happywork.fitur.presensi.model.faceDeetectionModel.FaceDetectionRootModel;
import com.pdamkotasmg.happywork.fitur.presensi.model.savePresensiModel.SavePresensiRootModel;
import com.pdamkotasmg.happywork.fitur.authentication.login.model.AkunRootModel;
import com.pdamkotasmg.happywork.fitur.dashboard.model.ShfitPegawaiRootModel;
import com.pdamkotasmg.happywork.fitur.feeds.model.BeritaRootModel;
import com.pdamkotasmg.happywork.fitur.kehadiran.model.RiwayatKehadiranRootModel;
import com.pdamkotasmg.happywork.fitur.perangkat.model.PerangkatRootModel;
import com.pdamkotasmg.happywork.fitur.splash.model.androidVersion.AndroidVersionModel;
import com.pdamkotasmg.happywork.fitur.splash.model.packageName.PackageNameRootModel;

import okhttp3.MultipartBody;
import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.Headers;
import retrofit2.http.Multipart;
import retrofit2.http.POST;
import retrofit2.http.Part;
import retrofit2.http.Query;

public interface ApiService {
    //    // google
//    @GET("siteverify")
//    Call<ResponseBody> getCaptcha(@Query("secret") String secretKey, @Query("response") String responseToken);
//
//    // akun
//    @FormUrlEncoded
//    @POST("auth-pelanggan/send-otp")
//    Call<AkunRootModel> sendOTP(
//            @Field("phone") String phone
//    );
//
    @FormUrlEncoded
    @POST("auth/login")
    Call<AkunRootModel> login(
            @Field("npp") String npp,
            @Field("password") String password,
            @Field("hwid") String hwid,
            @Field("model") String model,
            @Field("product") String product,
            @Field("device") String device,
            @Field("build_brand") String build_brand,
            @Field("os_version") String os_version,
            @Field("sdk_version") String sdk_version,
            @Field("build_number") String build_number,
            @Field("build_incremental") String build_incremental,
            @Field("ip_address") String ip_address,
            @Field("connection_type") String connection_type,
            @Field("ssid") String ssid,
            @Field("location_city") String location_city,
            @Field("latitude") String latitude,
            @Field("longitude") String longitude,
            @Field("app_version") String app_version
    );

    @POST("auth/logout")
    Call<ResponseBody> logout(@Header("Authorization") String auth);

    @POST("auth/delete-all-session")
    Call<ResponseBody> deleteAllSession(@Header("Authorization") String auth);

    @FormUrlEncoded
    @POST("attendance/check-location")
    Call<CheckLocationRootModel> checkLocation(@Header("Authorization") String auth,
                                               @Field("type") String statusPresensi,
                                               @Field("npp") String npp,
                                               @Field("latitude") double latitude,
                                               @Field("longitude") double longitude);

    @FormUrlEncoded
    @POST("attendance/record")
    Call<SavePresensiRootModel> savePresensi(@Header("Authorization") String auth,
                                           @Field("latitude") double latitude,
                                           @Field("longitude") double longitude,
                                           @Field("type") String statusPresensi,
                                           @Field("npp") String npp,
                                           @Field("check_face") String check_face,
                                           @Field("photo_path") String photo_path,
                                           @Field("connection_type") String connection_type
                                           );

    @GET("masterdata/forbid-app")
    Call<PackageNameRootModel> getPackageName();

    @GET("masterdata/config/all")
    Call<AndroidVersionModel> getAndroidVersion();

    @Headers({"Content-Type: application/json;charset=UTF-8"})
    @GET("auth/active-session")
    Call<PerangkatRootModel> getHistoryAktifDevice(@Header("Authorization") String auth);

    @Headers({"Content-Type: application/json;charset=UTF-8"})
    @GET("attendance/applies-shift")
    Call<ShfitPegawaiRootModel> getShiftPegawai(@Header("Authorization") String auth);


    //
//    @POST("auth-pelanggan/register")
//    Call<RegisterModel> register(
//            @Body RegisterModel registerModel
//    );
//
    @Headers({ "Content-Type: application/json;charset=UTF-8"})
    @GET("attendance/history/")
    Call<RiwayatKehadiranRootModel> getHistoryPresensi(
            @Header("Authorization") String auth,
            @Query("date_from") String dateFrom,
            @Query("date_to") String dateTo,
            @Query("formatted") String formatted
            );
//
//    @Headers({ "Content-Type: application/json;charset=UTF-8"})
//    @GET("auth-pelanggan/refresh-token")
//    Call<RefreshTokenRoot> refreshToken(@Header("Authorization") String auth);
//
    @GET("kontol")
    Call<BeritaRootModel> getNews();

    //
//    @Headers({ "Content-Type: application/json;charset=UTF-8"})
//    @GET("pengaduan/mobile/aduanku")
//    Call<PengaduanRootModel> getPengaduan(@Header("Authorization") String auth);
//
    @Multipart
    @POST("attendance/face-recognize")
    Call<FaceDetectionRootModel> checkFace(
            @Header("Authorization") String auth,
            @Part("type") RequestBody statusPresensi,
            @Part("npp") RequestBody npp,
            @Part MultipartBody.Part filePartPhoto
    );
//
//    @Headers({ "Content-Type: application/json;charset=UTF-8"})
//    @GET("pengaduan/mobile/daftar-kategori")
//    Call<KategoriRootModel> getKategori(@Header("Authorization") String auth);

}

File: app/src/main/java/com/pdamkotasmg/happywork/api/server/ApiService.kt
package com.pdamkotasmg.happywork.api.server

import com.pdamkotasmg.happywork.fitur.authentication.login.model.AkunRootModel
import com.pdamkotasmg.happywork.fitur.feeds.model.BeritaRootModel
import com.pdamkotasmg.happywork.fitur.perangkat.model.PerangkatRootModel
import com.pdamkotasmg.happywork.fitur.splash.model.androidVersion.AndroidVersionModel
import com.pdamkotasmg.happywork.fitur.splash.model.packageName.PackageNameRootModel
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.http.*

interface ApiService {
    //    // google
    //    @GET("siteverify")
    //    Call<ResponseBody> getCaptcha(@Query("secret") String secretKey, @Query("response") String responseToken);
    //
    //    // akun
    //    @FormUrlEncoded
    //    @POST("auth-pelanggan/send-otp")
    //    Call<AkunRootModel> sendOTP(
    //            @Field("phone") String phone
    //    );
    //
    @FormUrlEncoded
    @POST("auth/login")
    fun login(
            @Field("npp") npp: String?,
            @Field("password") password: String?,
            @Field("hwid") hwid: String?,
            @Field("model") model: String?,
            @Field("product") product: String?,
            @Field("device") device: String?,
            @Field("build_brand") build_brand: String?,
            @Field("os_version") os_version: String?,
            @Field("sdk_version") sdk_version: String?,
            @Field("build_number") build_number: String?,
            @Field("build_incremental") build_incremental: String?,
            @Field("ip_address") ip_address: String?,
            @Field("connection_type") connection_type: String?,
            @Field("ssid") ssid: String?,
            @Field("location_city") location_city: String?,
            @Field("latitude") latitude: String?,
            @Field("longitude") longitude: String?,
            @Field("app_version") app_version: String?
    ): Call<AkunRootModel>

    @POST("auth/logout")
    fun logout(@Header("Authorization") auth: String?): Call<ResponseBody?>?
    @POST("auth/delete-all-session")
    fun deleteAllSession(@Header("Authorization") auth: String?): Call<ResponseBody?>?
    @GET("masterdata/forbid-app")
    fun getPackageName(): Call<PackageNameRootModel?>?
    @GET("masterdata/config/all")
    fun getAndroidVersion(): Call<AndroidVersionModel?>?
    @Headers("Content-Type: application/json;charset=UTF-8")
    @GET("auth/active-session")
    fun getHistoryAktifDevice(@Header("Authorization") auth: String?): Call<PerangkatRootModel?>?

    //
    //    @POST("auth-pelanggan/register")
    //    Call<RegisterModel> register(
    //            @Body RegisterModel registerModel
    //    );
    //
    //    @Headers({ "Content-Type: application/json;charset=UTF-8"})
    //    @GET("billing/tagihan/air/{nolangg}")
    //    Call<BillingTagihanRootModel> getBillingTagihan(@Path("nolangg") String nolangg, @Header("Authorization") String auth);
    //
    //    @Headers({ "Content-Type: application/json;charset=UTF-8"})
    //    @GET("auth-pelanggan/refresh-token")
    //    Call<RefreshTokenRoot> refreshToken(@Header("Authorization") String auth);
    //
    @GET("portal-web/berita")
    fun getNews(): Call<BeritaRootModel?>? //
    //    @Headers({ "Content-Type: application/json;charset=UTF-8"})
    //    @GET("pengaduan/mobile/aduanku")
    //    Call<PengaduanRootModel> getPengaduan(@Header("Authorization") String auth);
    //
    //    @Multipart
    //    @POST("pengaduan/mobile")
    //    Call<PengaduanRootModel> sendPengaduanFix(
    //            @Header("Authorization") String auth,
    //            @Part("nama_pengadu") RequestBody nama_pengadu,
    //            @Part("uraian") RequestBody uraian,
    //            @Part("telp_pengadu") RequestBody telp_pengadu,
    //            @Part("alamat_pengadu") RequestBody alamat_pengadu,
    //            @Part("latitude") RequestBody latitude,
    //            @Part("longitude") RequestBody longitude,
    //            @Part MultipartBody.Part filePartAduan,
    //            @Part MultipartBody.Part filePartPengadu,
    //            @Part("kategori_aduan") RequestBody kategori
    //    );
    //
    //    @Headers({ "Content-Type: application/json;charset=UTF-8"})
    //    @GET("pengaduan/mobile/daftar-kategori")
    //    Call<KategoriRootModel> getKategori(@Header("Authorization") String auth);
}